packages/kbn-apm-config-loader/src/config.test.ts

```diff
@@ -41,6 +41,34 @@ describe('ApmConfiguration', () => {
     expect(config.getConfig('myservice').serviceVersion).toBe('9.2.1');
   });
 
+  it('allows overriding the service name via root config', () => {
+    const kibanaConfig = {
+      elastic: {
+        apm: {
+          serviceName: 'serviceNameOverride',
+        },
+      },
+    };
+    const config = new ApmConfiguration(mockedRootDir, kibanaConfig, true);
+    expect(config.getConfig('myservice').serviceName).toBe('serviceNameOverride');
+  });
+
+  it('allows overriding the service name via service override', () => {
+    const kibanaConfig = {
+      elastic: {
+        apm: {
+          servicesOverrides: {
+            myservice: {
+              serviceName: 'serviceNameOverride',
```

Comment:
After reading the linked issues, I understand where this request is coming from. However, I'm a bit concerned that users may choose to name all services (Elasticsearch, Kibana Server, and Kibana RUM) of the same stack (`project ABC`) the same way, and have mixed traces under the same service name.

Again, not an APM expert, but I think doing this might cause unexpected behavior. I wonder if the recommendation should be: keep the service name untouched but use global labels to identify stacks.

cc @elastic/apm-ui